nautobot/extras/api/serializers.py

```diff
@@ -877,6 +902,39 @@ def get_display(self, obj):
         return obj.created.isoformat()
 
 
+#
+# Notes
+#
+
+
+class NoteSerializer(BaseModelSerializer):
+    url = serializers.HyperlinkedIdentityField(view_name="extras-api:note-detail")
+    user = NestedUserSerializer(read_only=True)
+    assigned_object_type = ContentTypeField(queryset=ContentType.objects.all())
+    assigned_object = serializers.SerializerMethodField(read_only=True)
+
+    class Meta:
+        model = Note
+        fields = [
+            "url",
+            "user",
+            "user_name",
+            "assigned_object_type",
+            "assigned_object_id",
+            "assigned_object",
+            "note",
+            "slug",
+        ]
+
+    @extend_schema_field(serializers.DictField(allow_null=True))
+    def get_assigned_object(self, obj):
+        if obj.assigned_object is None:
+            return None
+        serializer = get_serializer_for_model(obj.assigned_object, prefix="Nested")
```

Comment:
Not sure if we need a try/except block here - what if we have a plugin model that supports Notes but doesn't implement a nested serializer, is it OK for this endpoint to return a 500 error in that case?

Reply:
I can return a better error, or have it None, which would you prefer?